lib/i18n/japanese.js

```diff
@@ -0,0 +1,121 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Genie
+//
+// Copyright 2019 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Giovanni Campagna <gcampagn@cs.stanford.edu>, Jian Li <jianli19@cs.stanford.edu>
+//
+// See COPYING for details
+"use strict";
+
+var keys = [
+];
+
+function postprocessSynthetic(sentence, program) {
+
+    keys.forEach((key) => {
+        var re = new RegExp("\\b" + key[0] + "\\b", "g");
+        if(sentence.match(re))
+            sentence = sentence.replace(key[0], key[1]);
+    });
+    return sentence;
+}
+
+// no overrides for Chinese, arguments should be translated properly in Thingpedia
+const ARGUMENT_NAME_OVERRIDES = {};
+
+
+const IGNORABLE_TOKENS = {
+    'sportradar': ['fc', 'ac', 'us', 'if', 'as', 'rc', 'rb', 'il', 'fk', 'cd', 'cf'],
+    'imgflip:meme_id': ['the'],
+    'tt:currency_code': ['us'],
+    'tt:stock_id': ['l.p.', 's.a.', 'plc', 'n.v', 's.a.b', 'c.v.'],
+    'org:freedesktop:app_id': ['gnome']
+};
+
+const ABBREVIATIONS = [
+    ['公司', '有限公司', '股份有限公司'],
+    ['高铁', '高速铁路'],
+    ['网路', '网络'],
+    ['&', '和'],
+];
```

Comment:
How did you compile this abbreviation list?
Wikipedia for example lists KK as a standard abbreviation for a stock-traded company: https://en.wikipedia.org/wiki/Kabushiki_gaisha and GK for a limited-liability company: https://en.wikipedia.org/wiki/G%C5%8Dd%C5%8D_gaisha

Reply:
I did not compile this. It comes from Chinese package. I will omit it for now.
The language packages are very primitive right now and check basic things.
I will start writing the tokenizers for these languages which should help me learn new things to gradually improve the lang packages too.